### PR TITLE
Replace manual composed grpc call with Broker methods in DataNode

### PR DESCRIPTION
**Pull request by congqixia**
See also #27675
/kind enhancement

### CODE REVIEW

File: internal/datanode/flow_graph_delete_node.go [hunk 1]
```diff
@@ -159,6 +159,7 @@ func (dn *deleteNode) bufferDeleteMsg(msg *msgstream.DeleteMsg, tr TimeRange, st
 
 	primaryKeys := storage.ParseIDs2PrimaryKeys(msg.PrimaryKeys)
 	segIDToPks, segIDToTss := dn.filterSegmentByPK(msg.PartitionID, primaryKeys, msg.Timestamps)
+	log.Warn("CQX filter result", zap.Any("segIDToPks", segIDToPks))
```
Comment (XuanYang-cn) [inline internal/datanode/flow_graph_delete_node.go:162]: Please remove this

File: internal/datanode/flow_graph_delete_node.go [hunk 2]
```diff
@@ -185,6 +186,7 @@ func (dn *deleteNode) filterSegmentByPK(partID UniqueID, pks []primaryKey, tss [
 	segments := dn.channel.filterSegments(partID)
 	for index, pk := range pks {
 		for _, segment := range segments {
+			log.Warn("CQX process segment pk", zap.Any("pk", pk), zap.Bool("result", segment.isPKExist(pk)))
```
Comment (XuanYang-cn) [inline internal/datanode/flow_graph_delete_node.go:189]: ditto

**Review by XuanYang-cn — CHANGES_REQUESTED** [review]

File: internal/datanode/flow_graph_delete_node.go (hunk 2 above)
Comment (congqixia) [reply internal/datanode/flow_graph_delete_node.go]: done

**Pull request merged**